docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,300 @@
+collaborate on the MicroPython project.
+
+Configure upstream if you cloned your fork.
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+
+It is common to configure ``upstream`` and ``origin`` on a forked repository
+to assist with sharing code changes. You can maintain your own mapping but
+it is recommended that ``origin`` maps to your fork and ``upstream`` to the main
+MicroPython repository:
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+   $ git remote set-url origin https://github.com/<your-user-name>/micropython.git
+
+After the above configuration, your setup should be similar to this:
+
+.. code-block:: console
+   
+   $ git remote -v
+   origin	https://github.com/<your-user-name>/micropython.git (fetch)
+   origin	https://github.com/<your-user-name>/micropython.git (push)
+   upstream	https://github.com/micropython/micropython.git (fetch)
+   upstream	https://github.com/micropython/micropython.git (push)
+
+You should now have a copy of the source code. By default you are pointing
+to the master branch. To prepare for further development, it is recommended
+to work on a development branch.
+
+.. code-block:: console
+
+    $ git checkout -b dev-branch
+
+You can give it any name. You will have to compile MicroPython whenever you change 
+to a different branch.
+
+Compile and build the Code
+--------------------------
+
+For development, you should compile MicroPython for a specific platform which corresponds
+to a port. Start by installing the required dependencies. Then build the MicroPython 
+cross-compiler before you can successfully compile and build.
+
+Required dependencies
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Before anything, make sure you have a working environment for ``gcc`` and ``GNU make``.
+Ubuntu 20.04 has been used for the example below but other unixes ought to work with
+little modification:
+
+.. code-block:: console
+
+   $ gcc --version
+   gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
+   Copyright (C) 2019 Free Software Foundation, Inc.
+   This is free software; see the source for copying conditions.  There is NO
+   warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+Install the required dependencies for linux:
+
+.. code-block:: console
+
+   $ sudo apt-get install build-essential libreadline-dev libffi-dev git \\
+   pkg-config gcc-arm-none-eabi libnewlib-arm-none-eabi
+
+For the stm32 port, the ARM cross-compiler is required:
+
+.. code-block:: console
+
+   $ sudo apt-get install arm-none-eabi-gcc arm-none-eabi-binutils arm-none-eabi-newlib
+
+Python 3 is also recommended:
+
+.. code-block:: console
+
+   $ python
+   Python 3.5.0 (default, Jul 17 2020, 14:04:10) 
+   [GCC 5.4.0 20160609] on linux
+   Type "help", "copyright", "credits" or "license" for more information.
+   >>> 
+
```

Comment:
I'd add a line here stating other ports will have different requirements, and that information is available in the README (at the moment, perhaps also in the docs, not sure)

Reply:
Initially, I approached this in a detail as needed basis i.e share dependency details under the docs for a given port. This is better I guess? than moving this in a separate section? because It looks like a lot of redundancy..in that we may need to have many notes to remind people later in the doc. Thoughts?

This applies to the `gcc` dependency too. It could have been better detailed under the Unix port. Those were my initial thoughts before addressing a review to move this but am open to thoughts.

Reply:
Not sure if I'm completely following, you mean the idea is to have individual port build documentation start with 'for dependencies and building the cross-compiler see gettingstarted.rst'? I think that's ok, but still I'd have this section clearly state which are the dependencies common to (almost) all ports: git, python3 and make+gcc a.k.a. build-essential (libffi is only for the unix port).

Anyway my comment was only slightly related to that: my point was that now if you read this section, it goes like
- install the required dependencies for linux
- install dependencies for the sm32 port

Which leaves the reader wondering 'ok but what about the other ports?'. So that's why i was suggesting an extra line saying other ports have othe requirements.